prow/config/config.go

```diff
@@ -213,6 +214,18 @@ type InRepoConfig struct {
 	AllowedClusters map[string][]string `json:"allowed_clusters,omitempty"`
 }
 
+type AllowedAPIClient struct {
+	Type              string              `json:"type,omitempty"`
+	ID                string              `json:"id,omitempty"`
```

Comment:
Note to self: just have a field `*GcpProject` and also another field `*AwsProject` or some such, inside a struct. That way, because they are pointers, the user only has to define one of them per element in a slice.

Of course we still have to do some extra validation to ensure that exactly 1 field is populated.